extensions/docker/packages/api/src/docker-extension-api.d.ts

```diff
@@ -0,0 +1,42 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+// handle the context information of the docker contexts
+// https://docs.docker.com/engine/manage-resources/contexts/
+export interface DockerContextInfo {
+  name: string;
+  isCurrentContext: boolean;
+  metadata: {
+    description: string;
+  };
+  endpoints: {
+    docker: {
+      host: string;
+    };
+  };
+}
+
+// omit current context as it is coming from another source
+// disabling the rule as we're not only extending the interface but omitting one field
+// but the rule is not able to understand that
+// eslint-disable-next-line @typescript-eslint/no-empty-object-type
+export interface DockerContextParsingInfo extends Omit<DockerContextInfo, 'isCurrentContext'> {}
```

Comment:
When using Omit, we usually use a type instead of an interface.

```suggestion
export type DockerContextParsingInfo  = Omit<DockerContextInfo, 'isCurrentContext'>
```

And remove the `eslint-disable-next-line @typescript-eslint/no-empty-object-type`

yet I am not sure to understand why you need to use Omit, you could define a middle interface without the isCurrentContext, and extend this interface to add the isCurrentContext.

Something like

````ts
interface BaseType {
  name: string;
  metadata: {
    description: string;
  };
  endpoints: {
    docker: {
      host: string;
    };
  };
}

interface ComplexType extends BaseType  {
  isCurrentContext: boolean;
}
````

The second proposition has some advantages: it still have two types, and you avoid using Omit synthax, which may be confusing, just need to find the naming!

Reply:
This is moved code

Reply:
Approving then